cmd/snap-confine/mount-support.c

```diff
@@ -207,6 +196,107 @@ static void sc_setup_mount_profiles(const char *snap_name)
 	}
 }
 
+// Find snap-update-ns, returns NULL if one cannot be found.
+static const char *find_snap_update_ns()
+{
+	// Note that at this stage we _may_ be after pivot_root but we cannot rely
+	// on this. If we did pivot_root then the base snap _may_ contain the tool
+	// but we, again, cannot rely on that.
+	const char *const reexec_tools[] = {
```

Comment:
Let's keep an fd open (to snap-update-ns) and exec it this way. Also use /proc/self/exe to find the tool before.